lib/src/services/ds_toast.service.dart

```diff
@@ -0,0 +1,364 @@
+      mainButton: mainButton,
+    );
+
+    Overlay.of(context)!.insert(_overlayEntry!);
+
+    _controlAnimation = Control.playFromStart;
+  }
+
+  OverlayEntry createOverlayEntry(
+      {required BuildContext context,
+      required String message,
+      int? animationDuration,
+      required int toastDuration,
+      Widget? icon,
+      Widget? mainButton,
+      String? title}) {
+    return OverlayEntry(
+      builder: (context) => Positioned(
+        bottom: 50.0 + positionOffset!,
+        width: MediaQuery.of(context).size.width - 16.0,
+        left: 8.0,
+        child: Dismissible(
+          key: const Key('key'),
+          onDismissed: (direction) {
+            if (_overlayEntry != null) {
+              _controlAnimation = Control.stop;
+              _close();
+            }
+          },
+          child: _animeCard(),
+        ),
+      ),
+    );
+  }
+
+  /// Create the toast card
+  Material _cardToast() {
+    return Material(
+      elevation: 10.0,
+      borderRadius: BorderRadius.circular(8.0),
+      child: Container(
+        decoration: BoxDecoration(
+          color: backgroundColor,
+          borderRadius: BorderRadius.circular(8.0),
+        ),
+        child: Row(
+          mainAxisAlignment: MainAxisAlignment.spaceAround,
+          crossAxisAlignment: CrossAxisAlignment.start,
+          children: <Widget>[
+            if (icon != null)
+              Container(
+                alignment: Alignment.topLeft,
+                child: icon,
+              ),
+            Flexible(
+              child: Padding(
+                padding: const EdgeInsets.all(16.0),
+                child: Column(
+                  crossAxisAlignment: CrossAxisAlignment.start,
+                  children: [
+                    if (title != null) DSHeadlineSmallText(text: title),
+                    SizedBox(
+                      child: DSBodyText(
+                        text: message,
+                        overflow: TextOverflow.visible,
+                      ),
+                    ),
+                  ],
+                ),
+              ),
+            ),
+            Container(
+              alignment: Alignment.topLeft,
+              child: _setMainButton(),
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+
+  /// Prepares the presentation of toast elements according to the type
+  void _prepareToast(final DSToastType type) {
+    switch (type) {
+      case DSToastType.success:
+        backgroundColor = DSColors.primaryGreensMint;
+        icon = Padding(
+          padding: const EdgeInsets.only(top: 20.0, left: 20.0),
+          child: _setIcon('svg', 'assets/images/icon_success.svg'),
+        );
+        break;
+      case DSToastType.warning:
+        backgroundColor = DSColors.primaryYellowsCorn;
+        icon = Padding(
+          padding: const EdgeInsets.only(top: 20.0, left: 20.0),
+          child: _setIcon('img', 'assets/images/icon_alert_warning.png'),
```

Comment:
I totally agree with using only SVG, but it's not an established standard and other widgets use PNG. What I found again I already created the SVG, but I used PNG for others where the image already existed. Creating SVG for everyone would lead to having duplicate images in assets. Better to leave this as a technical debt and we'll adjust everything in the future. This widget and the others.